### DISCUSSION TITLE
large Mat objects and release()

### DISCUSSION THREAD

**Discussion by ngbusca** [bytedeco/javacv#2109] [Q&A] [answered]
OpenCV is known for requiring the user to call `.release()` on Mat objects to avoid memory fill ups. The issue being that the garbage collector is not aware of the proper memory footprint of those objects.

Is this a similar requirement of JavaCV? If not, how does java CV contour this issue?

The demo code in the README says that one can call `deallocate()` if one wishes to, but will the garbage collector take care of releasing Mats in due time?

**Comment by saudet** [comment, marked as answer]
That's what PointerScope is for: http://bytedeco.org/news/2018/07/17/bytedeco-as-distribution/
